key.core/src/main/java/de/uka/ilkd/key/rule/metaconstruct/DeclareLocalLoopVars.java

```diff
@@ -0,0 +1,54 @@
+/* This file is part of KeY - https://key-project.org
+ * KeY is licensed under the GNU General Public License Version 2
+ * SPDX-License-Identifier: GPL-2.0-only */
+package de.uka.ilkd.key.rule.metaconstruct;
+
+import java.util.ArrayList;
+
+import de.uka.ilkd.key.java.Expression;
+import de.uka.ilkd.key.java.ProgramElement;
+import de.uka.ilkd.key.java.Services;
+import de.uka.ilkd.key.java.abstraction.ArrayType;
+import de.uka.ilkd.key.java.abstraction.KeYJavaType;
+import de.uka.ilkd.key.java.declaration.LocalVariableDeclaration;
+import de.uka.ilkd.key.java.declaration.VariableSpecification;
+import de.uka.ilkd.key.java.reference.TypeRef;
+import de.uka.ilkd.key.logic.op.LocationVariable;
+import de.uka.ilkd.key.logic.op.ProgramSV;
+import de.uka.ilkd.key.rule.inst.SVInstantiations;
+
+import org.key_project.util.collection.ImmutableMap;
+
+/**
+ * A transformer that takes a ProgramSV, which must be instantiated with a map from variable to
```

Comment:
Are ProgramSVs not instantiated by ProgramElements? How can that be matched to a map?

Reply:
Yeah, this is a bit of a hack. It would probably be better to directly create the necessary variable declarations and updates in \newLocalVars().